### ISSUE TITLE
💅 False positive for noUnusedPrivateClassMembers when method is used as a variable

### ISSUE THREAD

**Issue by profaurore** [biomejs/biome#8176]
### Environment information

```bash
CLI:
  Version:                      2.3.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.17.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.13.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  true
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementInteractions
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noForEach
    complexity/noImplicitCoercions
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringConcat
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    complexity/useSimplifiedLogicExpression
    complexity/useWhile
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalDirnameFilename
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNestedComponentDefinitions
    correctness/noNodejsModules
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noPrivateImports
    correctness/noProcessGlobal
    correctness/noReactPropAssignments
    correctness/noRenderReturnValue
    correctness/noRestrictedElements
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredDependencies
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useExhaustiveDependencies
    correctness/useGraphqlNamedOperations
    correctness/useHookAtTopLevel
    correctness/useIsNan
    correctness/useJsonImportAttributes
    correctness/useJsxKeyInIterable
    correctness/useParseIntRadix
    correctness/useSingleJsDocAsterisk
    correctness/useUniqueElementIds
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noFloatingPromises
    nursery/noShadow
    nursery/useExhaustiveSwitchCases
    nursery/useSortedClasses
    performance/noAccumulatingSpread
    performance/noAwaitInLoops
    performance/noBarrelFile
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noImgElement
    performance/noNamespaceImport
    performance/noReExportAll
    performance/noUnwantedPolyfillio
    performance/useGoogleFontPreconnect
    performance/useSolidForComponent
    performance/useTopLevelRegex
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noCommonJs
    style/noDefaultExport
    style/noDescendingSpecificity
    style/noDoneCallback
    style/noEnum
    style/noExportedImports
    style/noHeadElement
    style/noImplicitBoolean
    style/noInferrableTypes
    style/noNamespace
    style/noNegationElse
    style/noNestedTernary
    style/noNonNullAssertion
    style/noParameterAssign
    style/noParameterProperties
    style/noProcessEnv
    style/noRestrictedGlobals
    style/noRestrictedImports
    style/noRestrictedTypes
    style/noShoutyConstants
    style/noSubstr
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/noValueAtRule
    style/noYodaExpression
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useAtIndex
    style/useBlockStatements
    style/useCollapsedElseIf
    style/useCollapsedIf
    style/useComponentExportOnlyModules
    style/useConsistentArrayType
    style/useConsistentBuiltinInstantiation
    style/useConsistentCurlyBraces
    style/useConsistentMemberAccessibility
    style/useConsistentObjectDefinitions
    style/useConst
    style/useDefaultParameterLast
    style/useDefaultSwitchClause
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExplicitLengthCheck
    style/useExponentiationOperator
    style/useExportType
    style/useFilenamingConvention
    style/useForOf
    style/useFragmentSyntax
    style/useGroupedAccessorPairs
    style/useImportType
    style/useLiteralEnumMembers
    style/useNamingConvention
    style/useNodeAssertStrict
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useNumericSeparators
    style/useObjectSpread
    style/useReadonlyClassProperties
    style/useSelfClosingElements
    style/useShorthandAssign
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useSymbolDescription
    style/useTemplate
    style/useThrowNewError
    style/useThrowOnlyError
    style/useTrimStartEnd
    style/useUnifiedTypeSignatures
    suspicious/noAlert
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConsole
    suspicious/noConstEnum
    suspicious/noConstantBinaryExpressions
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDocumentImportInPage
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyBlockStatements
    suspicious/noEmptyInterface
    suspicious/noEvolvingTypes
    suspicious/noExplicitAny
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFocusedTests
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noHeadImportInDocument
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisplacedAssertion
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSkippedTests
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnassignedVariables
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useAwait
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useErrorMessage
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useGuardForIn
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword
    suspicious/useNumberToFixedDigitsArgument
    suspicious/useStaticResponseMethods
    suspicious/useStrictMode

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedPrivateClassMembers

### Playground link

https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAVABlAHMAdAAgAHsACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAKQAgAHsACgAgACAAIAAgAHQAaABpAHMALgB4ACAAPQAgAHUAcwBlAGQATQBlAHQAaABvAGQAOwAKACAAIAB9AAoACgAgACAAcAByAGkAdgBhAHQAZQAgAHUAcwBlAGQATQBlAHQAaABvAGQAKAApACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAAxADsACgAgACAAfQAKAH0A

### Expected result

For private methods used as variables to be considered used. In my specific scenario, I am binding the method as an event handler to the custom element class instance `this.addEventListener(this.myMethod.bind(this))`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I don't really understand the code that you shared, is there more to it that you forgot? 

You mentioned some event handlers, but I don't see anything like that in the playground.

It seems to me that the rule is correct in this case

**Comment by mafrasi2** [comment]
This appears to be a problem with all kinds of private class members, not just private functions.

I think [this playground](https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAVABlAHMAdAAgAHsACgAgACAAcAByAGkAdgBhAHQAZQAgAHUAcwBlAGQATQBlAG0AYgBlAHIAIAA9ACAAMQAKAAoAIAAgAGcAZQB0AEMAYQBsAGMAdQBsAGEAdABlAGQATgB1AG0AYgBlAHIAKAApADoAIABuAHUAbQBiAGUAcgAgAHsACgAgACAAIAAgAGwAZQB0ACAAeAAgAD0AIAB0AGgAaQBzAC4AdQBzAGUAZABNAGUAbQBiAGUAcgAKACAAIAAgACAALwAvACAAZABvACAAcwB0AHUAZgBmACAAdwBpAHQAaAAgAHgALgAuAC4ACgAgACAAIAAgAHgAIAAqAD0AIAAyAAoAIAAgACAAIAB4ACAAKwA9ACAAMwAKACAAIAAgACAALwAvACAAdwBlACAAYQByAGUAIABkAG8AbgBlAAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAHgACgAgACAAfQAKAAoAIAAgAHMAZQB0AFAAcgBpAHYAYQB0AGUATQBlAG0AYgBlAHIAKABuAGUAdwBNAGUAbQBiAGUAcgA6ACAAbgB1AG0AYgBlAHIAKQAgAHsACgAgACAAIAAgAHQAaABpAHMALgB1AHMAZQBkAE0AZQBtAGIAZQByACAAPQAgAG4AZQB3AE0AZQBtAGIAZQByAAoAIAAgAH0ACgB9AAoAYwBvAG4AcwB0ACAAdABlAHMAdAAgAD0AIABuAGUAdwAgAFQAZQBzAHQAKAApAAoAdABlAHMAdAAuAHMAZQB0AFAAcgBpAHYAYQB0AGUATQBlAG0AYgBlAHIAKAAxADAAKQAKAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoAHQAZQBzAHQALgBnAGUAdABDAGEAbABjAHUAbABhAHQAZQBkAE4AdQBtAGIAZQByACgAKQApAA%3D%3D) should be a bit simpler. It is actually quite hard to *not* trigger the rule.

**Comment by GeneZharov** [comment]
Playground [example](https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABDAFUAUwBUAE8ATQBfAEUAUgBSAE8AUgBfAE4AQQBNAEUAIAA9ACAAIgBDAHUAcwB0AG8AbQBFAHIAcgBvAHIAIgA7AAoACgBlAHgAcABvAHIAdAAgAGMAbABhAHMAcwAgAEMAdQBzAHQAbwBtAEUAcgByAG8AcgAgAGUAeAB0AGUAbgBkAHMAIABFAHIAcgBvAHIAIAB7AAoAIAAgAG8AdgBlAHIAcgBpAGQAZQAgAG4AYQBtAGUAIAA9ACAAQwBVAFMAVABPAE0AXwBFAFIAUgBPAFIAXwBOAEEATQBFADsACgAKACAAIAAvAC8AIABOAG8AZABlAC0AbABpAGsAZQAgAGUAcgByAG8AcgAgAGkAZABlAG4AdABpAGYAaQBlAHIACgAgACAAYwBvAGQAZQAgAD0AIAAiACIAOwAKAAoAIAAgAC8ALwAgAEgAdQBtAGEAbgAtAHIAZQBhAGQAYQBiAGwAZQAgAGUAcgByAG8AcgAgAG0AZQBzAHMAYQBnAGUACgAgACAALwAvACAALQAtAC0ALQAKACAAIAAvAC8AIABXAGgAeQAgAGAAbQBlAHMAcwBhAGcAZQBgACAAcAByAG8AcABlAHIAdAB5ACAAdwBhAHMAIABuAG8AdAAgAGUAbgBvAHUAZwBoADoACgAgACAALwAvACAAMQAuACAAUwBvAG0AZQB0AGkAbQBlAHMAIAB3AGUAIABuAGUAZQBkACAAdABvACAAcgBlAG4AZABlAHIAIABwAHUAcgBlACAAaAB1AG0AYQBuAC0AcgBlAGEAZABhAGIAbABlACAAdABlAHgAdAAgACgAdwBpAHQAaAAgAG4AbwAgAGMAbwBkAGUAKQAKACAAIAAvAC8AIAAyAC4AIABXAGUAIABkAG8AbgAnAHQAIAB3AGEAbgB0ACAAdABvACAAZAB1AHAAbABpAGMAYQB0AGUAIABlAHIAcgBvAHIAIABjAG8AZABlAHMAIABpAG4AIABhACAAYABtAGUAcwBzAGEAZwBlAGAAIABzAHQAcgBpAG4AZwAgAHcAaABlAG4ACgAgACAALwAvACAAIAAgAHMAZQByAGkAYQBsAGkAegBpAG4AZwAgAGEAbgBkACAAdABoAGUAbgAgAGQAZQBzAGUAcgBpAGEAbABpAHoAaQBuAGcAIABhACAAYwB1AHMAdABvAG0AIABlAHIAcgBvAHIAOgAKACAAIAAvAC8AIAAgACAAbgBlAHcAIABDAHUAcwB0AG8AbQBFAHIAcgBvAHIAKABzAGUAcgBpAGEAbABpAHoAZQBDAHUAcwB0AG8AbQBFAHIAcgBvAHIAKABjAHUAcwB0AG8AbQBFAHIAcgBvAHIAKQApAAoAIAAgAHQAZQB4AHQAIAA9ACAAIgAiADsACgAKACAAIABjAG8AbgBzAHQAcgB1AGMAdABvAHIAKABlAHIAcgA%2FADoAIABzAHQAcgBpAG4AZwApADsACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAZQByAHIAPwA6ACAAewAgAGMAbwBkAGUAPwA6ACAAcwB0AHIAaQBuAGcAOwAgAHQAZQB4AHQAPwA6ACAAcwB0AHIAaQBuAGcAIAB9ACkAOwAKACAAIABjAG8AbgBzAHQAcgB1AGMAdABvAHIAKABlAHIAcgA%2FADoAIAB7AAoAIAAgACAAIABuAGEAbQBlADoAIAB0AHkAcABlAG8AZgAgAEMAVQBTAFQATwBNAF8ARQBSAFIATwBSAF8ATgBBAE0ARQA7AAoAIAAgACAAIABjAG8AZABlADoAIABzAHQAcgBpAG4AZwA7AAoAIAAgACAAIAB0AGUAeAB0ADoAIABzAHQAcgBpAG4AZwA7AAoAIAAgACAAIABtAGUAcwBzAGEAZwBlADoAIABzAHQAcgBpAG4AZwA7AAoAIAAgAH0AKQA7AAoAIAAgAGMAbwBuAHMAdAByAHUAYwB0AG8AcgAoAGUAcgByAD8AOgAgAGEAbgB5ACkAIAB7AAoAIAAgACAAIABzAHUAcABlAHIAKAApADsACgAgACAAIAAgAGkAZgAgACgAdAB5AHAAZQBvAGYAIABlAHIAcgAgAD0APQA9ACAAIgBzAHQAcgBpAG4AZwAiACkAIAB7AAoAIAAgACAAIAAgACAAdABoAGkAcwAuAHQAZQB4AHQAIAA9ACAAdABoAGkAcwAuAG0AZQBzAHMAYQBnAGUAIAA9ACAAZQByAHIAOwAKACAAIAAgACAAfQAKACAAIAAgACAAaQBmACAAKAB0AHkAcABlAG8AZgAgAGUAcgByACAAPQA9AD0AIAAiAG8AYgBqAGUAYwB0ACIAIAAmACYAIABlAHIAcgAgACEAPQA9ACAAbgB1AGwAbAApACAAewAKACAAIAAgACAAIAAgAGMAbwBuAHMAdAAgAHsAIABjAG8AZABlACAAPQAgACIAIgAsACAAdABlAHgAdAAgAD0AIAAiACIALAAgAG0AZQBzAHMAYQBnAGUAIAA9ACAAIgAiACAAfQAgAD0AIABlAHIAcgA7AAoAIAAgACAAIAAgACAAdABoAGkAcwAuAGMAbwBkAGUAIAA9ACAAYwBvAGQAZQA7AAoAIAAgACAAIAAgACAAaQBmACAAKABlAHIAcgAuAG0AZQBzAHMAYQBnAGUAKQAgAHsACgAgACAAIAAgACAAIAAgACAAdABoAGkAcwAuAHQAZQB4AHQAIAA9ACAAdABlAHgAdAA7AAoAIAAgACAAIAAgACAAIAAgAHQAaABpAHMALgBtAGUAcwBzAGEAZwBlACAAPQAgAG0AZQBzAHMAYQBnAGUAOwAKACAAIAAgACAAIAAgAH0AIABlAGwAcwBlACAAewAKACAAIAAgACAAIAAgACAAIAB0AGgAaQBzAC4AdABlAHgAdAAgAD0AIAB0AGUAeAB0ADsACgAgACAAIAAgACAAIAAgACAAdABoAGkAcwAuAG0AZQBzAHMAYQBnAGUAIAA9ACAAdABoAGkAcwAuACMAZgBvAHIAbQBhAHQATQBlAHMAcwBhAGcAZQAoAGMAbwBkAGUALAAgAHQAZQB4AHQAKQA7AAoAIAAgACAAIAAgACAAfQAKACAAIAAgACAAfQAKACAAIAB9AAoACgAgACAALwAvACAARQByAHIAbwByACAAbQBlAHMAcwBhAGcAZQAgAGkAbgBjAGwAdQBkAGUAcwAgAGUAcgByAG8AcgAgAGMAbwBkAGUAIABiAGUAYwBhAHUAcwBlACAAcwB0AHIAaQBuAGcAaQBmAGkAZQBkACAAZQByAHIAbwByAHMAIABtAHUAcwB0ACAAYgBlAAoAIAAgAC8ALwAgAGEAcwAgAGkAbgBmAG8AcgBtAGEAdABpAHYAZQAgAGEAcwAgAHAAbwBzAHMAaQBiAGwAZQAuACAARQBzAHAAZQBjAGkAYQBsAGwAeQAgAGkAZgAgAGUAcgByAG8AcgAgAG0AZQBzAHMAYQBnAGUAIAB3AGEAcwAgAG4AbwB0ACAAcwBwAGUAYwBpAGYAaQBlAGQALgAKACAAIAAjAGYAbwByAG0AYQB0AE0AZQBzAHMAYQBnAGUAKABjAG8AZABlAD8AOgAgAHMAdAByAGkAbgBnACwAIABtAGUAcwBzAGEAZwBlAD8AOgAgAHMAdAByAGkAbgBnACkAOgAgAHMAdAByAGkAbgBnACAAewAKACAAIAAgACAAaQBmACAAKAAhAGMAbwBkAGUAIAAmACYAIABtAGUAcwBzAGEAZwBlACkAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIABtAGUAcwBzAGEAZwBlADsACgAgACAAIAAgAH0AIABlAGwAcwBlACAAaQBmACAAKABjAG8AZABlACAAJgAmACAAIQBtAGUAcwBzAGEAZwBlACkAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIABgAFsAJAB7AGMAbwBkAGUAfQBdAGAAOwAKACAAIAAgACAAfQAgAGUAbABzAGUAIABpAGYAIAAoAGMAbwBkAGUAIAAmACYAIABtAGUAcwBzAGEAZwBlACkAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIABgAFsAJAB7AGMAbwBkAGUAfQBdADoAIAAkAHsAbQBlAHMAcwBhAGcAZQB9AGAAOwAKACAAIAAgACAAfQAgAGUAbABzAGUAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIAAiACIAOwAKACAAIAAgACAAfQAKACAAIAB9AAoAfQA%3D) from my code base
It used to work fine in version 2.3.4

**Comment by profaurore** [comment]
@ematipico Sorry about that. I was trying to boil it down to the basics. Here is a [more complete example](https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAVABlAHMAdAAgAGUAeAB0AGUAbgBkAHMAIABIAFQATQBMAEUAbABlAG0AZQBuAHQAIAB7AAoAIAAgAGMAbwBuAHMAdAByAHUAYwB0AG8AcgAoACkAIAB7AAoAIAAgACAAIAB0AGgAaQBzAC4AYQBkAGQARQB2AGUAbgB0AEwAaQBzAHQAZQBuAGUAcgAoACcAYwBsAGkAYwBrACcALAAgAHQAaABpAHMALgB1AHMAZQBkAE0AZQB0AGgAbwBkAC4AYgBpAG4AZAAoAHQAaABpAHMAKQApADsACgAgACAAfQAKAAoAIAAgAHAAcgBpAHYAYQB0AGUAIAB1AHMAZQBkAE0AZQB0AGgAbwBkACgAKQAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAMQA7AAoAIAAgAH0ACgB9AA%3D%3D) from my project.

**Comment by SnowSquire** [comment]
> It used to work fine in version 2.3.4

This is due to https://github.com/biomejs/biome/pull/7684
In addition to it not tracking when methods are used as a variable, it doesn't account for static private members like `static #NTAB = 32`. since you access the member from the class's name and not `this`
[Playground](https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAVABlAHMAdAAgAHsACgAgACAAcAByAGkAdgBhAHQAZQAgAHMAdABhAHQAaQBjACAAdAB5AHAAZQBzAGMAcgBpAHAAdAAgAD0AIAAzADIACgAgACAAcwB0AGEAdABpAGMAIAAjAGUAYwBtAGEAIAA9ACAAMwAyAAoACgAgACAAcwBvAG0AZQB0AGgAaQBuAGcAKAB4ADoAIABuAHUAbQBiAGUAcgApACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAB4ACAAKgAgAFQAZQBzAHQALgB0AHkAcABlAHMAYwByAGkAcAB0ACAAKgAgAFQAZQBzAHQALgAjAGUAYwBtAGEACgAgACAAfQAKAH0A)


[It also doesn't track when private members get assigned to variables that are then used ](https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAUABhAGcAZQBNAGEAbgBhAGcAZQByADwAUABhAGcAZQBOAGEAbQBlACAAZQB4AHQAZQBuAGQAcwAgAHMAdAByAGkAbgBnACAAPQAgAHMAdAByAGkAbgBnAD4AIAB7AAoAIAAgACAAIAAjAHAAYQBnAGUAcwAgAD0AIAB7AH0AIABhAHMAIABSAGUAYwBvAHIAZAA8AFAAYQBnAGUATgBhAG0AZQAsACAASABUAE0ATABFAGwAZQBtAGUAbgB0AD4AOwAKAAkAYwB1AHIAcgBlAG4AdABQAGEAZwBlADoAIABQAGEAZwBlAE4AYQBtAGUAIAB8ACAAbgB1AGwAbAAgAD0AIABuAHUAbABsADsACgAJAAoACQBtAGEAawBlAFAAYQBnAGUAKABwAGEAZwBlAE4AYQBtAGUAOgAgAFAAYQBnAGUATgBhAG0AZQAsACAAcgBvAG8AdAA6ACAASABUAE0ATABFAGwAZQBtAGUAbgB0ACkAIAAgAHsACgAJAAkAdABoAGkAcwAuACMAcABhAGcAZQBzAFsAcABhAGcAZQBOAGEAbQBlAF0AIAA9ACAAcgBvAG8AdAA7AAoACQAJAHIAZQB0AHUAcgBuACAAcgBvAG8AdAA7AAoACQB9AAoACgAJAHMAaABvAHcAUABhAGcAZQAoAHAAYQBnAGUATgBhAG0AZQA6ACAAUABhAGcAZQBOAGEAbQBlACkAIAB7AAoACQAJAHQAaABpAHMALgBjAHUAcgByAGUAbgB0AFAAYQBnAGUAIAA9ACAAcABhAGcAZQBOAGEAbQBlADsACgAJAAkAYwBvAG4AcwB0ACAAcABhAGcAZQBSAG8AbwB0ACAAPQAgAHQAaABpAHMALgAjAHAAYQBnAGUAcwBbAHAAYQBnAGUATgBhAG0AZQBdADsACgAJAAkACgAJAAkAZABvAGMAdQBtAGUAbgB0AC4AZwBlAHQARQBsAGUAbQBlAG4AdABCAHkASQBkACgAIgBwAGEAZwBlAHMAIgApACEALgBpAG4AbgBlAHIASABUAE0ATAAgAD0AIAAiACIAOwAKAAkACQBpAGYAIAAoAHAAYQBnAGUAUgBvAG8AdAApACAAewAKAAkACQAJAGQAbwBjAHUAbQBlAG4AdAAuAGcAZQB0AEUAbABlAG0AZQBuAHQAQgB5AEkAZAAoACIAcABhAGcAZQBzACIAKQAhAC4AYQBwAHAAZQBuAGQAQwBoAGkAbABkACgAcABhAGcAZQBSAG8AbwB0ACkAOwAKAAkACQB9AAoACQB9AAoAfQA%3D)

**Comment by SnowSquire** [comment]
fixed in 2.3.7 it seems
